components/Ui/Link/UiLink.vue

```diff
@@ -1,60 +1,83 @@
 <template>
-  <component
-    :is="isNuxtLink ? 'NuxtLink' : 'a'"
+  <NuxtLink
     v-track-click="segment"
-    :href="isAnchor && url"
-    :to="isNuxtLink && url"
-    :style="hasLink && 'cursor:pointer'"
-    :rel="isExternal && 'noopener'"
-    :target="isExternal ? '_blank' : undefined"
+    :external="computedExternal"
+    :target="computedTarget"
     :title="title"
+    :to="url"
     @click="emit('click')"
     @mouseenter="$emit('mouseenter')"
   >
     <slot />
-  </component>
+  </NuxtLink>
 </template>
 
 <script setup lang="ts">
 import { CtaClickedEventProp } from "~/types/segment";
 
+type NuxtLinkTarget =
+  | "_blank"
+  | "_parent"
+  | "_self"
+  | "_top"
+  | (string & {})
+  | null;
+
 interface Props {
+  external?: boolean;
   segment?: CtaClickedEventProp | undefined;
+  target?: NuxtLinkTarget;
   title?: string;
-  url?: string;
+  url: string;
 }
 
 const props = withDefaults(defineProps<Props>(), {
+  external: undefined,
   segment: undefined,
+  target: undefined,
```

Comment:
The goal of the `target` prop here is to have the option to pass a the `target` value explicitly and have it been passed forward to `NuxtLink`.

Further down, in the `computedTarget` function we check if the `target` has been explicitly passed (not `undefined`).

If it has been defined, we pass that to `NuxtLink`.

Then we check for the other scenarios that concern us:

- `isExternalFromSameDomain`: `target="_self"`
- `isExternalFromDifferentDomain`: `target="_blank"`

Otherwise we do pass `undefined`, which won't render any `target` attribute in the final anchor's HTML which indeed, as you said, defaults to the same logic as with `target="_self"`.